### PR TITLE
feat: lambda logging rules

### PR DESCRIPTION
**Pull request by Kevinwochan**
[Serverless Nag Pack - lambda logging rules](https://github.com/cdklabs/cdk-nag/issues/1676)

First rule of many to check if this is in-line with the team's standards

### CODE REVIEW

File: test/rules/Lambda.test.ts [hunk 1]
```diff
@@ -351,4 +353,29 @@ describe('AWS Lambda', () => {
       validateStack(stack, ruleId, TestType.VALIDATION_FAILURE);
     });
   });
+
+  describe('LambdaLogging: Ensure that Lambda functions have a corresponding Log Group', () => {
+    const ruleId = 'LambdaLogging';
+    test('Noncompliance 1', () => {
+      new CfnFunction(stack, 'rFunction', {
```
Comment (dontirun) [inline test/rules/Lambda.test.ts:360]:
```suggestion
      new CfnFunction(stack, 'Function', {
```

The `r` is a CloudFormation naming convention which shouldn't be used in CDK. I mistakenly followed this pattern in some of the older tests.

File: test/rules/Lambda.test.ts [hunk 2]
```diff
@@ -351,4 +353,29 @@ describe('AWS Lambda', () => {
       validateStack(stack, ruleId, TestType.VALIDATION_FAILURE);
     });
   });
+
+  describe('LambdaLogging: Ensure that Lambda functions have a corresponding Log Group', () => {
```
Comment (dontirun) [inline test/rules/Lambda.test.ts:357]: General comment, also add some tests using the `L2` construct (if possible). There are some cases ([example](https://github.com/cdklabs/cdk-nag/blob/main/src/rules/opensearch/OpenSearchNoUnsignedOrAnonymousAccess.ts#L42)) where the CDK uses custom resources/alternative patterns to achieve the same functionality. Having tests that use L2 constructs help us catch some of these problems

File: src/rules/lambda/LambdaLogging.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+import { parse } from 'path';
+import { CfnResource, Stack } from 'aws-cdk-lib';
+import { CfnFunction } from 'aws-cdk-lib/aws-lambda';
+import { NagRuleCompliance } from '../../nag-rules';
+
+/**
+ * Ensure that Lambda functions have a corresponding Log Group
```
Comment (dontirun) [inline src/rules/lambda/LambdaLogging.ts:11]:
```suggestion
 * Lambda functions explicitly define their CloudWatch Log Groups
```

 Rule comments are declarative

**Review by dontirun — CHANGES_REQUESTED** [review]
Please add this to the [Additional Rules section](https://github.com/cdklabs/cdk-nag/blob/main/RULES.md#additional-rules) in the `RULES.md`file.

That section lists rules that are not currently part of any  included `NagPacks`

**Review by dontirun — CHANGES_REQUESTED** [review]
- All the rules that are being added need tests.
- The Serverless Pack needs its own section in the RULES file
- The Serverless Pack needs a test
- Rule info needs to be declarative 
-  Explanations needs to describe the benefit of doing what the rule tells you to do

**Comment by Kevinwochan** [comment]
I've written up rules for the rest of the pack.
Just thought i'd get this Lambda Log rule through first so i can make sure the rest of the serverless pack implementation aligns

File: src/rules/appsync/AppSyncTracing.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+import { parse } from 'path';
+import { CfnResource, Stack } from 'aws-cdk-lib';
+import { CfnGraphQLApi } from 'aws-cdk-lib/aws-appsync';
+import { NagRuleCompliance } from '../../nag-rules';
+
+/**
+ * Ensure that AppSync APIs have tracing enabled
```
[the remark this answers is not in the document]
Comment (Kevinwochan) [reply src/rules/appsync/AppSyncTracing.ts]: removed from this PR

File: src/packs/serverless.ts [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+  /**
+   * Check API Gateway Resources
+   * @param node the CfnResource to check
+   * @param ignores list of ignores for the resource
+   */
+  private checkApiGw(node: CfnResource) {
+    this.applyRule({
+      info: 'Ensure tracing is enabled',
+      explanation:
+        'Amazon API Gateway provides active tracing support for AWS X-Ray. Enable active tracing on your API stages to sample incoming requests and send traces to X-Ray.',
```
[the remark this answers is not in the document]
Comment (Kevinwochan) [reply src/packs/serverless.ts]: Removed from this PR

File: RULES.md [hunk 6]
```diff
@@ -702,6 +702,7 @@ A collection of community rules that are not currently included in any of the pr
 | Rule ID               | Cause                                                              | Explanation                                                                                                                                                                 |
 | --------------------- | ------------------------------------------------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | LambdaFunctionUrlAuth | The Lambda Function URL allows for public, unauthenticated access. | AWS Lambda Function URLs allow you to invoke your function via a HTTPS end-point, setting the authentication to NONE allows anyone on the internet to invoke your function. |
+| LambdaLogging | The Lambda Function does not define a corresponding log group | Ensure that Lambda functions have a corresponding Log Group |
```
[the remark this answers is not in the document]
Comment (Kevinwochan) [reply RULES.md]: Updated with the value of explicit logging

File: RULES.md [hunk 7]
```diff
@@ -702,6 +702,7 @@ A collection of community rules that are not currently included in any of the pr
 | Rule ID               | Cause                                                              | Explanation                                                                                                                                                                 |
 | --------------------- | ------------------------------------------------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | LambdaFunctionUrlAuth | The Lambda Function URL allows for public, unauthenticated access. | AWS Lambda Function URLs allow you to invoke your function via a HTTPS end-point, setting the authentication to NONE allows anyone on the internet to invoke your function. |
+| LambdaLogLevel | The Lambda Function does not define an explicit Log Level | For cost optimization purposes, you should explicitly define the required log level for cost effective storage of Lambda Logs. |
```
Comment (dontirun) [inline RULES.md:705]: This is for the wrong rule

File: test/rules/Lambda.test.ts [hunk 8]
```diff
@@ -351,4 +357,63 @@ describe('AWS Lambda', () => {
       validateStack(stack, ruleId, TestType.VALIDATION_FAILURE);
     });
   });
+
+  describe('LambdaLogLevel: Lambda functions have a explicit log level', () => {
```
Comment (dontirun) [inline test/rules/Lambda.test.ts:361]: It looks like these tests are for the wrong rule

**Review by dontirun — CHANGES_REQUESTED** [review]

**Pull request closed without merging**